addons/crm/models/res_partner.py

```diff
@@ -36,6 +38,10 @@ def default_get(self, fields):
         return rec
 
     def _compute_opportunity_count(self):
+        self.opportunity_count = 0
+        if not self.env['crm.lead'].check_access_rights('read', raise_exception=False):
+            return
```

Comment:
Well, if we set a specific group on the field, we might as well use `_has_group` to check the user belongs to the group instead.